### PR TITLE
feat: add radio `optionType` api to set radio option type

### PR DESCRIPTION
**Pull request by xrkffgg**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/24798

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add `optionType`  api to set Radio type in Radio.Group when use options.       |
| 🇨🇳 Chinese | 新增 `optionType` 来设置 Radio.Group使用 `options`时的 Radio 类型。           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/radio/demo/radiogroup-options.md](https://github.com/ant-design/ant-design/blob/feat-radio-com/components/radio/demo/radiogroup-options.md)
[View rendered components/radio/index.en-US.md](https://github.com/ant-design/ant-design/blob/feat-radio-com/components/radio/index.en-US.md)
[View rendered components/radio/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-radio-com/components/radio/index.zh-CN.md)

### CODE REVIEW

File: components/radio/demo/radiogroup-options.md [hunk 1]
```diff
@@ -56,16 +57,36 @@ class App extends React.Component {
...
           value={value3}
+          component="button"
+        />
+        <br />
+        <br />
+        <Radio.Group
+          options={optionsWithDisabled}
+          onChange={this.onChange4}
+          value={value4}
+          component="button"
```
Comment (afc163) [inline components/radio/demo/radiogroup-options.md:88]:
```suggestion
          component={Radio.Button}
```

Comment (xrkffgg) [reply components/radio/demo/radiogroup-options.md:88]: 我是想直接用字符串来代表类型，`button` 简单一些

Comment (afc163) [reply components/radio/demo/radiogroup-options.md:88]: 这样更精确，本来的默认值就是 `component={Radio}`

Comment (xrkffgg) [reply components/radio/demo/radiogroup-options.md:88]: 我先试试

Comment (xrkffgg) [reply components/radio/demo/radiogroup-options.md:88]: component 当个属性好弄。当组件，不太熟。明天再研究研究。😅😅😅

File: components/radio/group.tsx [hunk 2]
```diff
@@ -61,20 +63,20 @@ const RadioGroup: React.FC<RadioGroupProps> = props => {
         if (typeof option === 'string') {
           // 此处类型自动推导为 string
           return (
-            <Radio
+            <Component
```
Comment (xrkffgg) [inline components/radio/group.tsx:66]: @afc163 @zombieJ 

大佬帮忙看看，Radio.Button  没办法引，这样没起作用。

File: components/radio/demo/radiogroup-options.md (hunk 1 above)
Comment (zombieJ) [reply components/radio/demo/radiogroup-options.md]: 用户如果传自定义组件，这里其实会有点问题。我们没有对应的样式。

ref #12852

Comment (xrkffgg) [reply components/radio/demo/radiogroup-options.md]: 这个 功能 一直都被提起啊。。。

Comment (zombieJ) [reply components/radio/demo/radiogroup-options.md]: 嗯，当时关掉也是考虑 `children` 设置不同，这个 API 就矛盾了。其实基于那个在冲突的时候 warning 应该就行了。

Comment (xrkffgg) [reply components/radio/demo/radiogroup-options.md]: 那这样 直接用个 type 方便点？

Comment (zombieJ) [reply components/radio/demo/radiogroup-options.md]: 嗯，我比较推荐 type。 @afc163

**Comment by xrkffgg** [comment]
改成了 `optionType`

File: components/radio/group.tsx [hunk 3]
```diff
@@ -127,6 +134,7 @@ const RadioGroup: React.FC<RadioGroupProps> = props => {
 
 RadioGroup.defaultProps = {
   buttonStyle: 'outline' as RadioGroupButtonStyle,
+  optionType: 'default' as RadioGroupOptionType,
```
Comment (zombieJ) [inline components/radio/group.tsx:137]: 不应该有 default，只有 `options` 时才有用，如果是 `children` 设置的话，需要 warning 说 `optionType` 无效。

**Review by afc163 — APPROVED** [review]

**Pull request merged**